core/src/main/java/com/graphhopper/util/details/IntersectionDetails.java

```diff
@@ -0,0 +1,160 @@
+
+    private final EdgeExplorer crossingExplorer;
+    private final NodeAccess nodeAccess;
+    private final Weighting weighting;
+
+    public IntersectionDetails(Graph graph, Weighting weighting) {
+        super(INTERSECTION);
+
+        crossingExplorer = graph.createEdgeExplorer();
+        nodeAccess = graph.getNodeAccess();
+        this.weighting = weighting;
+    }
+
+    @Override
+    public boolean isEdgeDifferentToLastEdge(EdgeIteratorState edge) {
+        int toEdge = edgeId(edge);
+        if (toEdge != fromEdge) {
+            // Important to create a new map and not to clean the old map!
+            intersectionMap = new HashMap<>();
+
+            List<IntersectionValues> intersectingEdges = new ArrayList<>();
+
+            int baseNode = edge.getBaseNode();
+            EdgeIteratorState tmpEdge;
+
+            double startLat = nodeAccess.getLat(baseNode);
+            double startLon = nodeAccess.getLon(baseNode);
+
+
+            EdgeIterator edgeIter = crossingExplorer.setBaseNode(baseNode);
+            while (edgeIter.next()) {
+                tmpEdge = edgeIter.detach(false);
+
+                IntersectionValues intersectionValues = new IntersectionValues();
+                intersectionValues.bearing = calculateBearing(startLat, startLon, tmpEdge);
+                intersectionValues.in = edgeId(tmpEdge) == fromEdge;
+                intersectionValues.out = edgeId(tmpEdge) == edgeId(edge);
+                // The in edge is always false, this means that u-turns are not considered as possible turning option
+                intersectionValues.entry = !intersectionValues.in && Double.isFinite(weighting.calcEdgeWeightWithAccess(tmpEdge, false));
+
+                intersectingEdges.add(intersectionValues);
+            }
+
+            intersectingEdges.sort(null);
+
+            List<Integer> bearings = new ArrayList<>(intersectingEdges.size());
+            List<Boolean> entries = new ArrayList<>(intersectingEdges.size());
+
+            for (int i = 0; i < intersectingEdges.size(); i++) {
+                IntersectionValues intersectionValues = intersectingEdges.get(i);
+                bearings.add(intersectionValues.bearing);
+                entries.add(intersectionValues.entry);
+                if (intersectionValues.in) {
+                    intersectionMap.put("in", i);
+                }
+                if (intersectionValues.out) {
+                    intersectionMap.put("out", i);
+                }
+            }
+
+            intersectionMap.put("bearings", bearings);
+            intersectionMap.put("entries", entries);
+
+            fromEdge = toEdge;
+            return true;
+        }
+        return false;
+    }
+
+    private int calculateBearing(double startLat, double startLon, EdgeIteratorState tmpEdge) {
+        double latitude;
+        double longitude;
+        PointList wayGeo = tmpEdge.fetchWayGeometry(FetchMode.ALL);
+        if (wayGeo.size() <= 3) {
```

Comment:
To be honest, I don't remember why I chose 3 instead of 2. Maybe this is just a typo. I will change this to 2.

> And why not always pick the lat,lon from adjNode?

Edges can be long and we need to know the angle at the intersection. If the edge is 1km long, the angle might be very different from the angle at the intersection.